modules/objectdb_profiler/editor/objectdb_profiler_panel.cpp

```diff
@@ -0,0 +1,447 @@
+		return snapshot_cache.get(p_snapshot_file_name);
+	} else {
+		Ref<DirAccess> snapshot_dir = _get_and_create_snapshot_storage_dir();
+		if (snapshot_dir.is_null()) {
+			ERR_PRINT("Could not access ObjectDB Snapshot directory");
+			return nullptr;
+		}
+
+		String full_file_path = snapshot_dir->get_current_dir().path_join(p_snapshot_file_name) + ".odb_snapshot";
+
+		Error err;
+		Ref<FileAccess> snapshot_file = FileAccess::open(full_file_path, FileAccess::READ, &err);
+		if (err != OK) {
+			ERR_PRINT("Could not open ObjectDB Snapshot file: " + full_file_path);
+			return nullptr;
+		}
+
+		Vector<uint8_t> content = snapshot_file->get_buffer(snapshot_file->get_length()); // we want to split on newlines, so normalize them
+		if (content.is_empty()) {
+			ERR_PRINT("ObjectDB Snapshot file is empty: " + full_file_path);
+			return nullptr;
+		}
+
+		Ref<GameStateSnapshotRef> snapshot = GameStateSnapshot::create_ref(p_snapshot_file_name, content);
+		if (snapshot.is_valid()) {
+			// don't cache a null snapshot
+			snapshot_cache.insert(p_snapshot_file_name, snapshot);
+		}
+		return snapshot;
+	}
+}
+
+void ObjectDBProfilerPanel::show_snapshot(const String &p_snapshot_file_name, const String &p_snapshot_diff_file_name) {
+	clear_snapshot();
+
+	current_snapshot = get_snapshot(p_snapshot_file_name);
+	if (p_snapshot_diff_file_name != "none") {
+		diff_snapshot = get_snapshot(p_snapshot_diff_file_name);
+	} else {
+		diff_snapshot.unref();
+	}
+
+	_view_tab_changed(view_tabs->get_current_tab());
+}
+
+void ObjectDBProfilerPanel::_view_tab_changed(int p_tab_idx) {
+	// populating tabs only on tab changed because we're handling a lot of data,
+	// and the editor freezes for while if we try to populate every tab at once
+	SnapshotView *view = cast_to<SnapshotView>(view_tabs->get_current_tab_control());
+	GameStateSnapshot *snapshot = current_snapshot.is_null() ? nullptr : current_snapshot->ptr();
+	GameStateSnapshot *diff = diff_snapshot.is_null() ? nullptr : diff_snapshot->ptr();
+	if (snapshot != nullptr && !view->is_showing_snapshot(snapshot, diff)) {
+		view->show_snapshot(snapshot, diff);
+	}
+}
+
+void ObjectDBProfilerPanel::clear_snapshot() {
+	for (SnapshotView *view : views) {
+		view->clear_snapshot();
+	}
+	current_snapshot.unref();
+}
+
+void ObjectDBProfilerPanel::set_enabled(bool p_enabled) {
+	take_snapshot->set_text("Take ObjectDB Snapshot");
+	take_snapshot->set_disabled(!p_enabled);
+}
+
+void ObjectDBProfilerPanel::_snapshot_rmb(const Vector2 &p_pos, MouseButton p_button) {
+	if (p_button != MouseButton::RIGHT) {
+		return;
+	}
+	rmb_menu->clear(false);
+
+	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Rename")), "Rename Snapshot", RC_MENU_OPERATIONS::RENAME);
+	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Folder")), "Show In Folder", RC_MENU_OPERATIONS::SHOW_IN_FOLDER);
+	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Remove")), "Delete Snapshot", RC_MENU_OPERATIONS::DELETE);
```

Comment:
Use "in" instead of "In" (as stop words aren't capitalized) and make it localizable:

```suggestion
	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Rename")), TTR("Rename Snapshot"), RC_MENU_OPERATIONS::RENAME);
	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Folder")), TTR("Show in Folder"), RC_MENU_OPERATIONS::SHOW_IN_FOLDER);
	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Remove")), TTR("Delete Snapshot"), RC_MENU_OPERATIONS::DELETE);
```